pkg/manager/impl/metrics_manager.go

```diff
@@ -0,0 +1,682 @@
+			return err
+		}
+
+		// backened overhead
+		latestUpstreamNode := m.getLatestUpstreamNodeExecution("end-node",
+			workflow.Closure.CompiledWorkflow.Primary.Connections.Upstream, nodeExecutions)
+		if latestUpstreamNode != nil && !nodeExecution.Closure.UpdatedAt.AsTime().Before(latestUpstreamNode.Closure.UpdatedAt.AsTime()) {
+			*spans = append(*spans, createOperationSpan(latestUpstreamNode.Closure.UpdatedAt, nodeExecution.Closure.UpdatedAt, nodeTeardown))
+		}
+	}
+
+	return nil
+}
+
+// parseTaskExecution partitions the task execution into a collection of Categorical and Reference Spans which are
+// returned as a hierarchical breakdown of the task execution.
+func parseTaskExecution(taskExecution *admin.TaskExecution) *core.Span {
+	spans := make([]*core.Span, 0)
+
+	// check if plugin has started yet
+	if taskExecution.Closure.StartedAt == nil || reflect.DeepEqual(taskExecution.Closure.StartedAt, emptyTimestamp) {
+		spans = append(spans, createOperationSpan(taskExecution.Closure.CreatedAt, taskExecution.Closure.UpdatedAt, taskSetup))
+	} else {
+		// frontend overhead
+		spans = append(spans, createOperationSpan(taskExecution.Closure.CreatedAt, taskExecution.Closure.StartedAt, taskSetup))
+
+		// check if plugin has completed yet
+		if taskExecution.Closure.Duration == nil || reflect.DeepEqual(taskExecution.Closure.Duration, emptyDuration) {
+			spans = append(spans, createOperationSpan(taskExecution.Closure.StartedAt, taskExecution.Closure.UpdatedAt, taskRuntime))
+		} else {
+			// plugin execution
+			taskEndTime := timestamppb.New(taskExecution.Closure.StartedAt.AsTime().Add(taskExecution.Closure.Duration.AsDuration()))
+			spans = append(spans, createOperationSpan(taskExecution.Closure.StartedAt, taskEndTime, taskRuntime))
+
+			// backend overhead
+			if !taskExecution.Closure.UpdatedAt.AsTime().Before(taskEndTime.AsTime()) {
+				spans = append(spans, createOperationSpan(taskEndTime, taskExecution.Closure.UpdatedAt, taskTeardown))
+			}
+		}
+	}
+
+	return &core.Span{
+		StartTime: taskExecution.Closure.CreatedAt,
+		EndTime:   taskExecution.Closure.UpdatedAt,
+		Id: &core.Span_TaskId{
+			TaskId: taskExecution.Id,
+		},
+		Spans: spans,
+	}
+}
+
+// parseTaskExecutions partitions the task executions into a collection of Categorical and Reference Spans which are
+// appended to the provided spans argument.
+func parseTaskExecutions(taskExecutions []*admin.TaskExecution, spans *[]*core.Span, depth int) {
+	// sort task executions
+	sort.Slice(taskExecutions, func(i, j int) bool {
+		x := taskExecutions[i].Closure.CreatedAt.AsTime()
+		y := taskExecutions[j].Closure.CreatedAt.AsTime()
+		return x.Before(y)
+	})
+
+	// iterate over task executions
+	for index, taskExecution := range taskExecutions {
+		if index > 0 {
+			*spans = append(*spans, createOperationSpan(taskExecutions[index-1].Closure.UpdatedAt, taskExecution.Closure.CreatedAt, nodeReset))
```

Comment:
Is this efficient?
Is it equivalent to:
```golang
newSpans := *spans
...
newSpans = append(newSpans, ...)
...
...
*spans = newSpans
```

Reply:
Yes, everything I've read says this is the same efficiency.